src/lib/y2network/interface.rb

```diff
@@ -16,17 +16,75 @@
 #
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
+require "yast"
+
+require "forwardable"
+require "y2network/hwinfo"
+
+Yast.import "NetworkInterfaces"
+
 module Y2Network
   # Network interface.
+  #
+  # name and hardware attributes should be imutable for the whole life of the object.
+  # In other words - we have two configurations 1) system configuration - the one
+  # which is currently used by running system and 2) yast configuration - the one which
+  # is being modified an will be applied to the system on user's request. name and
+  # hardware should always reflect the first configuration bcs it is the only connection /
+  # identifier we can use when touching the system (e.g. when setting interfaces down, etc.)
   class Interface
     # @return [String] Device name (eth0, wlan0, etc.)
-    attr_accessor :name
+    attr_reader :name # TODO: when implementing renaming over new backend modifying name has to be checked
+    attr_accessor :renamed_to # TODO: use in renaming workflow
```

Comment:
I prefer something like system_name and name. This way you can also identify newly created interface.

Reply:
yes, as already described ... we differ only in variable naming ;-)